ts/components/MemberListItem.tsx

```diff
@@ -86,25 +116,204 @@ export const MemberListItem = (props: {
+
+  // we can always remove/and readd a non-admin member. So we consider that a member who accepted the invite cannot be resent an invite.
+  const canResendInvite = !acceptedInvite;
+
+  const shouldShowResendButton = canResendInvite || canResendPromotion;
+
+  if (!shouldShowResendButton) {
+    return null;
+  }
+  return (
+    <SessionButton
+      dataTestId={'resend-invite-button'}
+      buttonShape={SessionButtonShape.Square}
+      buttonType={SessionButtonType.Solid}
+      text={window.i18n('resend')}
+      onClick={async () => {
+        const group = await UserGroupsWrapperActions.getGroup(groupPk);
+        const member = await MetaGroupWrapperActions.memberGet(groupPk, pubkey);
+        if (!group || !group.secretKey || isEmpty(group.secretKey) || !member) {
+          window.log.warn('tried to resend invite but we do not have correct details');
+          return;
+        }
+        // if we tried to invite that member as admin right away, let's retry it as such.
+        const inviteAsAdmin = member.nominatedAdmin;
+        await GroupInvite.addJob({
+          groupPk,
+          member: pubkey,
+          inviteAsAdmin,
+          forceUnrevoke: true,
+        });
+      }}
+    />
+  );
+};
+
+const PromoteButton = ({ groupPk, pubkey }: { pubkey: PubkeyType; groupPk: GroupPubkeyType }) => {
+  const memberAcceptedInvite = useMemberHasAcceptedInvite(pubkey, groupPk);
+  const memberIsNominatedAdmin = useMemberIsNominatedAdmin(pubkey, groupPk);
+  const memberIsPendingRemoval = useMemberPendingRemoval(pubkey, groupPk);
+  // When invite-as-admin was used to invite that member, the resend button is available to resend the promote message.
+  // We want to show that button only to promote a normal member who accepted a normal invite but wasn't promoted yet.
+  // ^ this is only the case for testing. The UI will be different once we release the promotion process
+  if (
+    !hasClosedGroupV2QAButtons() ||
+    !memberAcceptedInvite ||
+    memberIsNominatedAdmin ||
+    memberIsPendingRemoval
+  ) {
+    return null;
+  }
+  return (
+    <SessionButton
+      dataTestId={'resend-promote-button'}
+      buttonShape={SessionButtonShape.Square}
+      buttonType={SessionButtonType.Solid}
+      buttonColor={SessionButtonColor.Danger}
+      text={window.i18n('promote')} // TODO DO NOT MERGE Remove after QA
+      onClick={() => {
+        void promoteUsersInGroup({
+          groupPk,
+          toPromote: [pubkey],
+        });
+      }}
+    />
+  );
+};
+
+export const MemberListItem = ({
+  isSelected,
+  pubkey,
+  dataTestId,
+  disableBg,
+  displayGroupStatus,
+  inMentions,
+  isAdmin,
+  isZombie,
+  onSelect,
+  onUnselect,
+  groupPk,
+  disabled,
+  withBorder,
+  maxNameWidth,
+  hideRadioButton,
+}: MemberListItemProps) => {
   const memberName = useNicknameOrProfileNameOrShortenedPubkey(pubkey);
+  const isUs = UserUtils.isUsFromCache(pubkey);
```

Comment:
Could we use `useOurPkStr` here?

Reply:
idk, we'd need a isUs hook specifically here, we can deal with that another time.